### PR TITLE
feat: snapmirror from source side [zapi changes]

### PR DESCRIPTION
**Pull request by Hardikl**

### CODE REVIEW

**Comment by Hardikl** [comment]
Initially I have started to add `snapmirror-get-destination-iter` zapi for solving the issue of missing `source_cluster`. While implementation, found that it could be handled just by `vserver-peer-get-iter` zapi and this zapi anyway we would need for handling peer vserver's alias conversion.

File: conf/zapi/cdot/9.8.0/snapmirror.yaml [hunk 1]
```diff
@@ -33,6 +33,9 @@ counters:
     - ^destination-location
     - ^policy-type
 
+input_param:
+  - expand: true
```
[the remark this answers is not in the document]
Comment (Hardikl) [reply conf/zapi/cdot/9.8.0/snapmirror.yaml:37]: no, it's from our side to get all granular records of SMs

File: cmd/collectors/zapi/collector/zapi.go [hunk 2]
```diff
@@ -169,6 +170,14 @@ func (z *Zapi) InitCache() error {
 		} else {
 			z.Logger.Trace().Msg("using default no batch-size")
 		}
+
+		z.inputParams = make(map[string]string)
+		if inputParams := z.Params.GetChildS("input_param"); inputParams != nil {
+			for _, inputParam := range inputParams.GetChildren() {
+				z.inputParams[inputParam.GetNameS()] = inputParam.GetContentS()
+			}
+			z.Logger.Debug().Msgf("using %d input params", len(z.inputParams))
```
Comment (cgrinds) [inline cmd/collectors/zapi/collector/zapi.go:179]: z.Logger.Debug().Int("len", len(z.inputParams)).Msg("using input params")

File: cmd/collectors/zapi/plugins/volume/volume.go [hunk 3]
```diff
@@ -376,7 +376,7 @@ func (my *Volume) getAggrDiskMapping() (map[string]aggrData, error) {
 	}
 
 	for _, aggrDiskData := range result {
-		aggrUUID := aggrDiskData.GetChildContentS("aaggregate-uuid")
+		aggrUUID := aggrDiskData.GetChildContentS("aggregate-uuid")
```
Comment (cgrinds) [inline cmd/collectors/zapi/plugins/volume/volume.go:379]: :0 nice catch

File: conf/zapi/cdot/9.8.0/snapmirror.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hardikl) [reply conf/zapi/cdot/9.8.0/snapmirror.yaml]: removed

File: cmd/collectors/zapi/collector/zapi.go [hunk 4]
```diff
@@ -380,6 +394,11 @@ func (z *Zapi) PollData() (map[string]*matrix.Matrix, error) {
 		if z.batchSize != "" {
 			request.NewChildS("max-records", z.batchSize)
 		}
+		if len(z.inputParams) != 0 {
+			for name, value := range z.inputParams {
+				request.NewChildS(name, value)
+			}
+		}
```
[the remark this answers is not in the document]
Comment (Hardikl) [reply cmd/collectors/zapi/collector/zapi.go]: done based on SM query

File: cmd/collectors/zapi/collector/zapi.go (hunk 2 above)
Comment (Hardikl) [reply cmd/collectors/zapi/collector/zapi.go]: code has been removed as not to expose new input param in ZAPI.

**Review by cgrinds — APPROVED** [review]

**Pull request merged**